core/src/main/java/org/elasticsearch/node/internal/InternalSettingsPreparer.java

```diff
@@ -114,10 +114,9 @@
             }
             if (loadFromEnv) {
                 for (String allowedSuffix : ALLOWED_SUFFIXES) {
-                    try {
-                        settingsBuilder.loadFromPath(environment.configFile().resolve("elasticsearch" + allowedSuffix));
-                    } catch (SettingsException e) {
-                        // ignore
+                    Path path = environment.configFile().resolve("elasticsearch" + allowedSuffix);
+                    if (Files.exists(path)) {
+                        settingsBuilder.loadFromPath(path);
                     }
                 }
```

Comment:
I'm wondering if we should fail if we find several times the config file with different extensions?